pkg/web/web.go

```diff
@@ -168,16 +172,35 @@ func New(ctx context.Context, opts ...Option) (*Server, error) {
 		server: server,
 	}
 
-	var staticDir http.Dir
+	var staticPath string
 	for _, path := range options.staticSearchPaths {
 		if s, err := os.Stat(path); err == nil && s.IsDir() {
-			staticDir = http.Dir(path)
+			staticPath = path
 			break
 		}
 	}
-	if staticDir != "" {
-		logger.WithFields(log.Fields("path", staticDir, "mount", options.staticMount)).Debug("Serving static assets")
+	if staticPath != "" {
+		staticDir := http.Dir(staticPath)
+		logger := logger.WithFields(log.Fields("path", staticDir, "mount", options.staticMount))
+		logger.Debug("Serving static assets")
 		s.Static(options.staticMount, staticDir, middleware.Immutable)
+
+		// register hashed filenames
+		manifest, err := ioutil.ReadFile(filepath.Join(staticPath, "manifest.yaml"))
+		if err != nil {
+			logger.WithError(err).Warn("Failed to load manifest.yaml")
+			return s, nil
+		}
+		hashedFiles := make(map[string]string)
+		err = yaml.Unmarshal(manifest, &hashedFiles)
+		if err != nil {
+			logger.WithError(err).Warn("Corrupted manifest.yaml")
```

Comment:
Shouldn't this error just be (wrapped and) returned?

Reply:
Not sure if that is what we want. Will that break development builds by chance?

If we are sure that it does not, I am fine with wrapping and returning the error.

Reply:
I'd say that if the file exists, it must be valid yaml.

Reply:
Hmm, yeah if the file exists then it should be valid. Updated the code.